### ISSUE TITLE
Single-sided cursors don't work in Windows

### ISSUE THREAD

**Issue by siakouchine** [plotly/plotly.js#2942]
Hello,
In parallel plots, It would be nice if we could have 2 different cursors when you drag a constraint range and when you resize it as it is implemented here: https://codepen.io/monfera/pen/apWqgM where a crosshair cursor is defined when you drag constraint ranges.
Thanks,
Stephane Iakouchine

**Comment by alexcjohnson** [comment]
We do, don't we? We use `n-resize` and `s-resize` for resizing the top and bottom, and `ns-resize` for dragging it. Is that not showing up for you?

![parcoords cursors](https://user-images.githubusercontent.com/2678795/44547879-6d3be500-a6ea-11e8-9484-309d90375736.gif)

**Comment by siakouchine** [comment]
Hello Alex,
Thanks for your very quick answer. 
Perhaps I do not use the last up to date version of Plotly. If you have a look on https://plot.ly/javascript/parallel-coordinates-plot/ demos, ns-resize is used for both resizing and dragging. These demos are using https://cdn.plot.ly/plotly-latest.min.js. 
Could you please tell me from where I can download the version you are using?
Many Thanks again.

**Comment by alexcjohnson** [comment]
I am using the latest version, and at https://plot.ly/javascript/parallel-coordinates-plot/ I see the same behavior shown in my gif above. Perhaps it's a browser issue? I was under the impression that all relevant browsers supported these cursors https://developer.mozilla.org/en-US/docs/Web/CSS/cursor but maybe not? What browser and OS are you using?

**Comment by siakouchine** [comment]
Mainly, I am using both Chrome (Version 68.0.3440.106 (Official Build) (64-bit)),  Microsoft Edge 38.14393.2068.0 and Firefox 52.7.2 (32 bits) browsers on Windows 10 (see attached video) : 
[PlotlyCursors.zip](https://github.com/plotly/plotly.js/files/2324772/PlotlyCursors.zip)

**Comment by alexcjohnson** [comment]
Wow, look at that... Windows doesn't seem to allow any of the single-sided cursors. Thanks for bringing this up @siakouchine! If I visit https://developer.mozilla.org/en-US/docs/Web/CSS/cursor on Browserstack, seems like all browsers and all Windows versions just silently use the double-sided resize cursors, even though the computed styles still show the single-sided version. There's this rather unhelpful note:
> In some environments, an equivalent bidirectional resize cursor is shown. For example, `n-resize` and `s-resize` are the same as `ns-resize`.

I guess "some environments" includes all of Windows 😞 Note that this impacts cartesian axes as well, where on x axes the left and right ends are supposed to be `w-resize` and `e-resize` respectively. Shape resizing also uses the single-sided cursors but I don't think the effect is ambiguous.

So... how to fix this? I'd like to continue supporting all three where they're available, but I don't see any way to tell what cursor *actually* gets rendered, and just saying "change behavior in Windows" doesn't seem robust. I suppose we could switch `ns-resize` to `row-resize` (and `ew-resize` to `col-resize`)... so it would still indicate the motion direction, with the (meaningless to us) bars in the middle showing that something changed. Not ideal, as the bars imply that something else is going on, but it would at least show when you moved from one region to the next.

@etpinard thoughts?

**Comment by etpinard** [comment]
> @etpinard thoughts?

Is there a way to polyfill cursors css?

> I suppose we could switch ns-resize to row-resize (and ew-resize to col-resize)... so it would still indicate the motion direction, with the (meaningless to us) bars in the middle showing that something changed

That's an idea. Perhaps, we could make could only apply that change if

```js
navigator.platform.indexOf('Win') === 0
```

?

**Comment by alexcjohnson** [comment]
> Perhaps, we could make could only apply that change if
> 
> ```js
> navigator.platform.indexOf('Win') === 0
> ```

Plausible. I had commented:

> just saying "change behavior in Windows" doesn't seem robust

But if we can convince ourselves that this is really the right criterion - ie it's *just* windows that omits the single-sided versions - I'd be OK with it.

I suppose we could also try to [make our own cursors](https://codepen.io/netsi1964/details/DsAhE) but that seems to have some compatibility issues so probably not a good idea for now.

**Comment by etpinard** [comment]
> But if we can convince ourselves that this is really the right criterion - ie it's just windows that omits the single-sided versions - I'd be OK with it.

Ideally, I wouldn't want to make our cursors worse-off in OSs other than Windows (e.g. using `row-resize` instead of `ns-resize`) just to make cursors in Windows look _ok_.

**Comment by alexcjohnson** [comment]
Heh to add to the confusion a little bit more - on my Mac, Chrome and Safari use different cursors for everything we're discussing here, but FF has the same cursors for `ns-resize` and `row-resize` (both have the line across the middle) and for `ew-resize` and `col-resize`. But I see (in Browserstack anyway, would be nice if someone could confirm this natively at https://developer.mozilla.org/en-US/docs/Web/CSS/cursor) that FF on Windows (tested with FF61 on Windows 7 and 10) has different cursors for these four even though `(n|s|ns)-resize` all look the same there.

Doesn't change the conclusion of how to fix the issue, but it's interesting (and frustrating) to notice that these inconsistencies don't depend purely on OS nor purely on browser but the combination of the two. At least I haven't (yet) seen any dependence on browser or OS version, at least across those we support.

So to be clear: the fix is to switch `ns-resize` to `row-resize` and `ew-resize` to `col-resize` on Windows only. If anyone on a different OS has contravening data please post it here! (@etpinard does Linux behave like a Mac?)

**Comment by etpinard** [comment]
On Ubuntu 18.04 

- in Chrome 68:

![peek 2018-08-29 09-46](https://user-images.githubusercontent.com/6675409/44791816-86241a80-ab70-11e8-9036-a4a20fafeb52.gif)

- in FF 61:

![peek 2018-08-29 09-48](https://user-images.githubusercontent.com/6675409/44791884-afdd4180-ab70-11e8-8cd7-229ea45fe8d4.gif)

**Comment by etpinard** [comment]
> So to be clear: the fix is to switch ns-resize to row-resize and ew-resize to col-resize on Windows only. 

:+1:

**Comment by siakouchine** [comment]
Hello Plotly team, do you have any news about this issue?

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $5k-$10k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
